lang/py/src/avro/io.py

```diff
@@ -280,6 +292,61 @@ def read_utf8(self):
     """
     return unicode(self.read_bytes(), "utf-8")
 
+  def read_date_from_int(self):
+    """
+    Decode python date object from int
+    int stores the number of days from
+    the unix epoch, 1 January 1970 (ISO calendar).
+    """
+    days_since_epoc = self.read_int()
+    return datetime.date(1970, 1, 1) + datetime.timedelta(days_since_epoc)
+
+  def _build_time_object(self, value, unit_multiplier):
+    value = value * (1000000 // unit_multiplier)
```

Comment:
I see. I think, similar to that, we could think of `unit_multiplier` as a value that converts the given value to micro-seconds, and simplify this to `value = value * scale_to_micro`